Pulsarc/Utils/InputManager.cs

```diff
@@ -1,123 +1,118 @@
+                            if (PressedKeys.Contains(key)) continue;
+                            KeyboardPresses.Enqueue(new KeyValuePair<double, Keys>(AudioManager.GetTime(), key));
+                            PressedKeys.Add(key);
 
-                        if (KeyboardState.GetPressedKeys().Count() > 0)
-                            foreach (Keys key in KeyboardState.GetPressedKeys())
+                            if (key == Keys.CapsLock)
                             {
-                                if (!PressedKeys.Contains(key))
-                                {
-                                    KeyboardPresses.Enqueue(new KeyValuePair<double, Keys>(AudioManager.GetTime(), key));
-                                    PressedKeys.Add(key);
-
-                                    if (key == Keys.CapsLock)
-                                    {
-                                        CapsLock = !CapsLock;
-                                        Caps = CapsLock;
-                                    }
-
-                                    if (key == Keys.LeftShift || key == Keys.RightShift)
-                                        Caps = !CapsLock;
-                                }
+                                CapsLock = !CapsLock;
+                                Caps = CapsLock;
                             }
 
-                        for (int i = 0; i < PressedKeys.Count; i++)
-                        {
-                            Keys key = PressedKeys[i];
+                            if (key == Keys.LeftShift || key == Keys.RightShift)
+                                Caps = !CapsLock;
+                        }
 
-                            if (!KeyboardState.IsKeyDown(key))
-                            {
-                                // Used if LN handling
-                                //keyboardReleases.Enqueue(new KeyValuePair<double, Keys>(time, key));
-                                PressedKeys.RemoveAt(i);
-                                i--;
+                    for (int i = 0; i < PressedKeys.Count; i++)
+                    {
+                        Keys key = PressedKeys[i];
 
+                        if (KeyboardState.IsKeyDown(key)) continue;
+                        // Used if LN handling
+                        //keyboardReleases.Enqueue(new KeyValuePair<double, Keys>(time, key));
+                        PressedKeys.RemoveAt(i);
+                        i--;
 
-                                if (key == Keys.LeftShift || key == Keys.RightShift)
-                                    Caps = CapsLock;
-                            }
-                        }
 
-                        if (LastMouseClick.Value == 0 && MouseState.LeftButton == ButtonState.Pressed)
-                        {
+                        if (key == Keys.LeftShift || key == Keys.RightShift)
+                            Caps = CapsLock;
+                    }
+
+                    switch (LastMouseClick.Value)
```

Comment:
I like the original if statements better, they're a bit easier to read imo, especially with all the other if statements around.
Read them out-loud, which one sounds nicer: 
"switch last mouse click dot value case 0 when mouse state dot left button equals button state dot pressed..." 
vs.
"if last mouse click dot value equals 0 and mouse state dot left button equals button state dot pressed..."

It'd probably be better if it were a nested block actually
```
if (LeftButtonPressed)
{
    if (LeftMouseClick.Value == 0)
        ...
    else
        ...
}
```